contracts/modules/capital/SafeTracker.sol

```diff
@@ -0,0 +1,159 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+pragma solidity ^0.8.18;
+
+import "../../interfaces/ISafeTracker.sol";
+import "../../interfaces/IPool.sol";
+import "../../interfaces/ISwapOperator.sol";
+import "../../interfaces/IPriceFeedOracle.sol";
+import "../../abstract/MasterAwareV2.sol";
+import "@openzeppelin/contracts-v4/token/ERC20/IERC20.sol";
+
+
+contract SafeTracker is ISafeTracker, MasterAwareV2 {
+
+  address public _swapOperator;
+  uint public investedUSDC;
+  uint public transferAmount;
+
+  string public symbol = "NXMIS";
+  string public name = "NXMIS";
+  uint8 public decimals = 18;
+
+  address public immutable safe;
+  IERC20 public immutable aweth;
+  IERC20 public immutable usdc;
+  IERC20 public immutable debtUsdc;
+
+  modifier onlyPoolAndSwapOperator {
+    require(msg.sender == internalContracts[uint(ID.P1)] || msg.sender == _swapOperator, "SafeTracker: Not Pool nor SwapOperator");
+    _;
+  }
+
+  modifier onlyPool {
+    require(msg.sender == internalContracts[uint(ID.P1)], "SafeTracker: Not Pool");
+    _;
+  }
+
+  modifier onlySafe {
+    require(msg.sender == safe, "SafeTracker: Not GnosisSafe");
+    _;
+  }
+
+  modifier onlySwapOperator {
+    require(msg.sender == internalContracts[uint(ID.P1)], "SafeTracker: Not SwapOperator");
+    _;
+  }
+
+  /* ========== CONSTRUCTOR ========== */
+
+  constructor(address _safe, address _usdc, address _aweth, address _debtUsdc) {
+    safe = _safe;
+    usdc = IERC20(_usdc);
+    aweth = IERC20(_aweth);
+    debtUsdc = IERC20(_debtUsdc);
+  }
+
+  /**
+  * @dev Gets the balance of the safe
+  * @return An uint256 representing the amount of the safe.
+  */
+  function totalSupply() external view onlyPool returns (uint256) {
+    return _calculateBalance();
+  }
+
+  /**
+  * @dev Gets the balance of the safe
+  * @return An uint256 representing the amount of the safe.
+  */
+  function balanceOf(address) external view onlyPool returns (uint256) {
+    return _calculateBalance();
+  }
```

Comment:
1. Anyone should be able to call `balanceOf`, no need for the modifier.
2. It should report that balance only for the pool contract address